beefy-cli/src/cli/mmr.rs

```diff
@@ -55,6 +56,13 @@ impl Mmr {
 				};
 				let leaf: Vec<u8> = Decode::decode(&mut leaf_content)?;
 				let leaf: MmrLeaf<u32, H256, H256> = Decode::decode(&mut &*leaf)?;
+				let expected_leaf_vers = LeafVersion::get();
+				if leaf.version != expected_leaf_vers {
```

Comment:
I'm wondering if it makes sense to compare the version components separately and refuse even printing the leaf if the major part does not match (that would require partially decoding the leaf first though). but perhaps it's an overkill.